packages/ui/components/scheduler/state/store.ts

```diff
@@ -0,0 +1,82 @@
+import create from "zustand";
+
+import dayjs from "@calcom/dayjs";
+
+import {
+  SchedulerComponentProps,
+  SchedulerPublicActions,
+  SchedulerState,
+  SchedulerStoreProps,
+} from "../types/state";
+import { mergeOverlappingDateRanges, weekdayDates } from "../utils";
+
+const defaultState: SchedulerComponentProps = {
+  view: "week",
+  startDate: weekdayDates(0, new Date()).startDate,
+  endDate: weekdayDates(0, new Date()).endDate,
+  events: [],
+  startHour: 0,
+  endHour: 23,
+  gridCellsPerHour: 4,
+};
+
+export const useSchedulerStore = create<SchedulerStoreProps>((set) => ({
+  ...defaultState,
+  setView: (view: SchedulerComponentProps["view"]) => set({ view }),
+  setStartDate: (startDate: SchedulerComponentProps["startDate"]) => set({ startDate }),
+  setEndDate: (endDate: SchedulerComponentProps["endDate"]) => set({ endDate }),
+  setEvents: (events: SchedulerComponentProps["events"]) => set({ events }),
+  // This looks a bit odd but init state only overrides the public props + actions as we don't want to override our internal state
+  initState: (state: SchedulerState & SchedulerPublicActions) => {
+    // Handle sorting of events if required
+    let events = state.events;
+
+    if (state.sortEvents) {
+      events = state.events.sort((a, b) => a.start.getTime() - b.start.getTime());
+    }
```

Comment:
Events need to be sorted - but if you don't pass in it sorted we offer a way to sort them on the fly